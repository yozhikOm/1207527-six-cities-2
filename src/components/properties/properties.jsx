import React, {PureComponent} from "react";
import PropTypes from 'prop-types';
import {PropertyCard} from "../property-card/property-card.jsx";

class Properties extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      hoveredCardID: -1
    };
    this._cardMouseEnterHandler = this._cardMouseEnterHandler.bind(this);
  }

  render() {
    return (
      <div className="cities__places-list places__list tabs__content">
        {this.props.items.map((item) => (
          <React.Fragment key={item.id}>
            <PropertyCard offerInfo={item} cardMouseEnterHandler={this._cardMouseEnterHandler} />
          </React.Fragment>
        ))}
      </div>
    );
  }

  _cardMouseEnterHandler(id) {
    this.setState({hoveredCardID: id});
  }
}

Properties.propTypes = {
  items: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        location: PropTypes.shape({
          city: PropTypes.string.isRequired,
          coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
        }).isRequired,
        title: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        description: PropTypes.string.isRequired,
        photos: PropTypes.arrayOf(
            PropTypes.shape({
              src: PropTypes.string,
            })
        ),
        host: PropTypes.string.isRequired,
      }).isRequired
  ),
};

export {Properties};
