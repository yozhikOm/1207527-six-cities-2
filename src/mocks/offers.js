export const offers = [
  {
    "id": 1,
    "title": `Cozy house amsterdam`,
    "type": `House`,
    "price": 650,
    "photos": [
      {
        "src": `img/apartment-01.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.85309666406198],
  },
  {
    "id": 2,
    "title": `Cosy room in spacious apartment`,
    "type": `Room`,
    "price": 45,
    "photos": [
      {
        "src": `img/apartment-02.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.369553943508, 4.85309666406198],
  },
  {
    "id": 3,
    "title": `Elegant private room in stylish house`,
    "type": `Room`,
    "price": 128,
    "photos": [
      {
        "src": `img/room.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.929309666406198],
  },
  {
    "id": 4,
    "title": `Huge Tinyhouse in the green, between sea and city`,
    "type": `House`,
    "price": 1520,
    "photos": [
      {
        "src": `img/apartment-03.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3809553943508, 4.939309666406198],
  }
];
