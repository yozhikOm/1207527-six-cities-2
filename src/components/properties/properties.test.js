import React from 'react';
import {Properties} from './properties.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {offers} from '../../mocks/offers.js';

it(`Properties компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  const mockCity = offers[0].city;

  renderer.render(<Properties
    offers={offers}
    setActiveItem={jest.fn()}
    currentCity={{
      title: mockCity.name,
      coordinates: [mockCity.location.latitude, mockCity.location.longitude]
    }}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
