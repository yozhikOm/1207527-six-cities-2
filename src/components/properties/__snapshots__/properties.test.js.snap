// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties компонент рендерится корректно 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          88
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        Some title
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Room
    </p>
  </div>
</article>
`;
